src/sentry/snuba/metrics/query.py

```diff
@@ -55,6 +55,13 @@ def __str__(self) -> str:
         metric_name = get_public_name_from_mri(self.metric_mri)
         return f"{self.op}({metric_name})" if self.op else metric_name
 
+    def __eq__(self, other: object) -> bool:
+        # The equal method is called after the hash method to verify for equality of objects to insert
+        # into the set. Because by default "__eq__()" does use the "is" operator we want to override it and
+        # model MetricField's equivalence as having the same hash value, in order to reuse the comparison logic defined
+        # in the "__hash__()" method.
+        return bool(self.__hash__() == other.__hash__())
```

Comment:
I have added `bool()` for the type checking in place in the CI.

Reply:
can you please add small test to demonstrate this specific problem ... it's hard to understand from `test_validate_order_by_field_in_select_with_different_alias`.

I mean create a test to compare 2 MetricField's